packages/@aws-cdk/aws-neptune/lib/cluster.ts

```diff
@@ -360,15 +369,15 @@ export abstract class DatabaseClusterBase extends Resource implements IDatabaseC
 
   protected abstract enableIamAuthentication?: boolean;
 
-  public grantConnect(grantee: iam.IGrantable): iam.Grant {
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
     if (this.enableIamAuthentication === false) {
-      throw new Error('Cannot grant connect when IAM authentication is disabled');
+      throw new Error('Cannot grant actions as IAM authentication is not enabled');
```

Comment:
tbh, I found the existing implementation of `grantConnect` and its side effect of possibly enabling iam auth, and how enabling iamAuth is handled a bit surprising!

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-neptune/lib/cluster.ts#L368

In terms of the error message, now that I look at it, I think the older version that uses "disabled" actually makes more sense and is more consistent with the implemented behaviour which
1. Defaults to disabled (I think a cloudformation default rather than a cdk default)
2. Requires one to explicitly disable iam auth if they don't want it and want grant methods to throw such exceptions

I get that `grant` enabling iam auth might be a convenience, but it's a bit twisted IMHO!

So, for this PR, how about
1. change the error message to `Cannot grant permissions when IAM authentication is disabled`
2. update the README to say `// Optional - will be automatically set if you call grant() or grantConnect()`